src/components/sbb-menu/sbb-menu.tsx

```diff
@@ -0,0 +1,297 @@
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_will-open',
+  })
+  public willOpen: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu is opened.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_did-open',
+  })
+  public didOpen: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu begins the closing transition.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_will-close',
+  })
+  public willClose: EventEmitter<void>;
+
+  /**
+   * Emits whenever the menu is closed.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+    eventName: 'sbb-menu_did-close',
+  })
+  public didClose: EventEmitter<void>;
+
+  /**
+   * Opens the menu on trigger click.
+   */
+  @Method()
+  public async open(): Promise<void> {
+    this.willOpen.emit();
+    this._setMenuPosition();
+    this._dialog.showModal();
+  }
+
+  /**
+   * Closes the menu.
+   */
+  @Method()
+  @Listen('sbb-menu-action_click')
+  public async close(): Promise<void> {
+    this.willClose.emit();
+    this._isDismissing = true;
+  }
+
+  // Resets position on window resize.
+  @Listen('resize', { target: 'window', passive: true })
+  public onResizeWindowEvent(): void {
+    if (this._open) {
+      this._setMenuPosition();
+    }
+  }
+
+  // Closes the menu on "Esc" key pressed.
+  @Listen('keydown', { target: 'window' })
+  public onEscAction(event: KeyboardEvent): void {
+    if (this._open && event.key === 'Escape') {
+      event.preventDefault();
+      this.close();
+    }
+  }
+
+  // Removes trigger click listener on trigger change.
+  @Watch('trigger')
+  public removeTriggerClickListener(
+    newValue: string | HTMLElement,
+    oldValue: string | HTMLElement
+  ): void {
+    if (newValue !== oldValue) {
+      this._menuController?.abort();
+      this._configure(this.trigger);
+    }
+  }
+
+  private _dialog: HTMLDialogElement;
+  private _triggerEl: HTMLElement;
+  private _pointerDownEventTarget: string | null;
+  private _menuController: AbortController;
+
+  public componentDidLoad(): void {
+    this._dialog = this.el.shadowRoot.querySelector('dialog');
+
+    // validate trigger element and attach event listeners
+    this._configure(this.trigger);
+
+    // close menu on interactive elements click
+    this._dismissOnInteractiveElementClick();
+
+    // listen for menu animation to end
+    this._dialog.onanimationend = (event: AnimationEvent) => this._onMenuAnimationEnd(event);
+  }
+
+  public disconnectedCallback(): void {
+    this._menuController.abort();
+  }
+
+  // Check if the trigger is valid and attach click event listeners.
+  private _configure(trigger: string | HTMLElement): void {
+    if (!trigger) {
+      return;
+    }
+
+    // check whether it's a string or an HTMLElement
+    if (typeof trigger === 'string') {
+      this._triggerEl = document.getElementById(trigger);
+    } else if (trigger instanceof window.Element) {
```

Comment:
Nitpick
```suggestion
    } else if (trigger instanceof HTMLElement) {
```

Reply:
I changed it to `window.HTMLElement` because otherwise I get this error:
`Rollup: Missing Export: ./src/components/sbb-menu/sbb-menu.tsx:1:131
           'HTMLElement' is not exported by
           ./node_modules/@stencil/core/internal/hydrate/index.js,
           imported by ./src/components/sbb-menu/sbb-menu.tsx`

Changing `HTMLElement` to `window.HTMLElement` quickly fix this.

Reply:
We did not find a solution but added a TODO, maybe you know more? :)